app/src/main/scala/org/alephium/explorer/api/EventsEndpoints.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.api
+
+import scala.collection.immutable.ArraySeq
+
+import sttp.tapir._
+import sttp.tapir.generic.auto._
+
+import org.alephium.api.{alphJsonBody => jsonBody}
+import org.alephium.explorer.api.BaseEndpoint
+import org.alephium.explorer.api.model._
+import org.alephium.protocol.model.TransactionId
+
+trait EventsEndpoints extends BaseEndpoint with QueryParams {
+
+  private val eventsEndpoint =
+    baseEndpoint
+      .tag("Smart contract events")
+      .in("smart-contract-events")
+
+  val getEventsByTxId: BaseEndpoint[TransactionId, ArraySeq[Event]] =
+    eventsEndpoint.get
+      .in("transaction-id")
+      .in(path[TransactionId]("transaction_id"))
+      .out(jsonBody[ArraySeq[Event]])
+      .description("Get contract events by transaction id")
+
+  val getEventsByContractAddress: BaseEndpoint[Address, ArraySeq[Event]] =
+    eventsEndpoint.get
+      .in("contract-address")
+      .in(path[Address]("contract_address"))
+      .out(jsonBody[ArraySeq[Event]])
+      .description("Get contract events by contract id")
+
+  val getEventsByContractAndInputAddress: BaseEndpoint[(Address, Address), ArraySeq[Event]] =
+    eventsEndpoint.get
+      .in("contract-address")
+      .in(path[Address]("contract_address"))
+      .in("input-address")
+      .in(path[Address]("input_address"))
+      .out(jsonBody[ArraySeq[Event]])
+      .description("Get contract events by contract id")
```

Comment:
Should "Get contract events by contract id" be "Get contract events by contract id and input address"?

Just for my knowledge: Are "contract id" and "contract address" the same thing?

Reply:
nice catch, I'll fix it.

Yes contract id and contract address is the same, maybe I should rather use `contract-id` everywhere.

Reply:
I think I'll rather use contract-address everywhere, as it's the notation received from the node https://github.com/alephium/alephium/blob/1e359e155b37c2afda6011cdc319d54ae8e4c059/api/src/main/scala/org/alephium/api/model/ContractEvents.scala#L43-L85

Reply:
done